### PR TITLE
updated model configs, fine-tuning recipe prams

### PR DESCRIPTION
**Pull request by malay-nagda**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

updated perf configs for different models targeted for h100, b200.
updated perf params for llama3 fine-tuning recipes.

**Collection**: [llm]

# Changelog

- Add specific line by line info of high level changes in this PR.

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

**Comment by guyueh1** [comment]
The if-condition to decide `enable_cg` in the code is becoming very unclean, I think we should just put a `cg` column in the csv and set it for each line. @erhoo82 what you think?

**Comment by erhoo82** [comment]
I agree. The if condition sounds quite random.
Let's use the csv config file.

File: nemo/collections/llm/recipes/llama31_405b.py [hunk 1]
```diff
@@ -397,10 +396,21 @@ def finetune_performance_optimizations(
             )
         )
     else:
+        recipe.trainer.strategy.account_for_embedding_in_pipeline_split = True
```
Comment (vysarge) [inline nemo/collections/llm/recipes/llama31_405b.py:399]: It looks like for pretrain asymmetric pipelining will be used regardless of performance_mode. Do we need to have it only for perf mode here?

Comment (malay-nagda) [reply nemo/collections/llm/recipes/llama31_405b.py:399]: We can have it in the main base recipe. But, does that need additional convergence testing as well?

**Comment by malay-nagda** [comment]
> I agree. The if condition sounds quite random. Let's use the csv config file.

added it as csv config. can you review @erhoo82?
@guyueh1 I've added cuda graphs values for gb200 too. Can you confirm if I set them for all required models?

File: nemo/collections/llm/recipes/llama31_405b.py (hunk 1 above)
Comment (malay-nagda) [reply nemo/collections/llm/recipes/llama31_405b.py]: updated in base recipe.

**Review by erhoo82 — DISMISSED** [review]

**Review by erhoo82 — APPROVED** [review]
